common/views.py

```diff
@@ -129,3 +144,117 @@ def post(self, request, *args, **kwargs):
                 messages.error(request, "Invalid token!")
                 return redirect("portal:home")
         return render(request, self.template_name, {"form": form})
+
+
+# @method_decorator(login_required, name="post")
```

Comment:
This should be enough, I believe, to cover both `get` and `post`
```suggestion
@method_decorator(login_required, name="dispatch")
```